smart-contracts/scripts/lock/clone.js

```diff
@@ -0,0 +1,80 @@
+const { ethers } = require('hardhat')
+const createLock = require('../deployments/lock')
+const listManagers = require('./listManagers')
+
+async function main({
+  lockAddress,
+  unlockAddress,
+  unlockVersion,
+  serializerAddress,
+}) {
+  const PublicLock = await ethers.getContractFactory('PublicLock')
+  const lock = PublicLock.attach(lockAddress)
+
+  // eslint-disable-next-line no-console
+  console.log(`CLONE LOCK > serializing: ${await lock.name()}...`)
+
+  const Serializer = await ethers.getContractFactory('LockSerializer')
+  const serializer = Serializer.attach(serializerAddress)
+  const serializedLock = await serializer.serialize(lockAddress)
+
+  // create the new lock
+  const newLockAddress = await createLock({
+    serializedLock,
+    unlockAddress,
+    unlockVersion,
+    salt: web3.utils.randomHex(12),
+  })
+
+  // eslint-disable-next-line no-console
+  console.log('CLONE LOCK > add key owners...')
+  const newLock = PublicLock.attach(newLockAddress)
+  const { keyOwners, expirationTimestamps, keyManagers } = serializedLock
+  const keyTx = await newLock.grantKeys(
```

Comment:
We probably need to paginate here. I worry this won't work for very large locks (with more than 100 members)

Reply:
That's a good question and i don't know. I think the lock with most users on xdai is this one https://blockscout.com/xdai/mainnet/address/0xac1fceC2e4064CCd83ac8C9B0c9B8d944AB0D246/transactions

We should try with it and see what happens